test/regression/riscv/instructions/compressed.cc

```diff
@@ -0,0 +1,692 @@
+    li x8, 4
+    c.jalr x8
+    mv t2, ra
+    addi t4, t4, 3
+  )");
+  EXPECT_EQ(getGeneralRegister<uint64_t>(30), 5);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(31), 10);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(29), 3);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(5), 20);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(6), 4);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(7), 12);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(1), 12);
+}
+
+TEST_P(InstCompressed, beqz) {
+  // Compressed branch if rs1 equal to zero
+  RUN_RISCV_COMP(R"(
+    addi x8, x8, 2
+    c.beqz x8, b1
+    addi x10, x10, 10
+    li x9, 0
+    c.beqz x9, b2
+    j b3
+    b1:
+    addi x10, x10, 5
+    b2:
+    addi x11, x11, 10
+    j b4
+    b3:
+    addi x11, x11, 5
+    b4:
+  )");
+  EXPECT_EQ(getGeneralRegister<uint64_t>(10), 10);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(11), 10);
+}
+
+TEST_P(InstCompressed, bnez) {
+  // Compressed branch if rs1 not equal to zero
+  RUN_RISCV_COMP(R"(
+    addi x8, x8, 0
+    c.bnez x8, b1
+    addi x10, x10, 10
+    li x9, 2
+    c.bnez x9, b2
+    j b3
+    b1:
+    addi x10, x10, 5
+    b2:
+    addi x11, x11, 10
+    j b4
+    b3:
+    addi x11, x11, 5
+    b4:
+  )");
+}
+
+TEST_P(InstCompressed, li) {
+  // Compressed load immediate
+  RUN_RISCV_COMP(R"(
+    addi a5, a5, 12
+    c.li a5, 0
+    addi a4, a4, 12
+    c.li a4, -32
+    addi a3, a3, 12
+    c.li a3, 31
+  )");
+  EXPECT_EQ(getGeneralRegister<uint64_t>(15), 0);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(14), -32);
+  EXPECT_EQ(getGeneralRegister<int64_t>(13), 31);
+}
+
+TEST_P(InstCompressed, lui) {
+  // Compressed load immediate into bits 17-12, clear bottom 12 and sign extend
+  // high bits
+  RUN_RISCV_COMP(R"(
+      c.lui t3, 4
+      c.lui t4, 0xFFFFC
+  )");
+  EXPECT_EQ(getGeneralRegister<uint64_t>(28), 4 << 12);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(29), -4ull << 12);
+}
+
+TEST_P(InstCompressed, addi) {
+  // Compressed add immediate
+  RUN_RISCV_COMP(R"(
+    c.addi t3, 3
+    c.addi t4, 6
+    c.addi t3, 30
+    c.addi zero, 16
+  )");
+  EXPECT_EQ(getGeneralRegister<uint64_t>(29), 6u);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(28), 33u);
+  EXPECT_EQ(getGeneralRegister<uint64_t>(0), 0);
+}
+
+TEST_P(InstCompressed, addiw) {
+  // Compressed add immediate. Produces 32 bit result and sign extends
+  RUN_RISCV_COMP(R"(
+    addi t3, t3, 91
+    slli t3, t3, 28
+    addiw t5, t3, -5
+    addiw t6, t2, -5
+  )");
+  EXPECT_EQ(getGeneralRegister<uint64_t>(28), 24427626496);
+  EXPECT_EQ(getGeneralRegister<int32_t>(30), -1342177285);
+  EXPECT_EQ(getGeneralRegister<int64_t>(31), -5);
+}
+
+TEST_P(InstCompressed, addi16sp) {
+  // Add immediate (multiple of 16) to stack pointer
+  RUN_RISCV_COMP(R"(
+    mv x8, sp
+    c.addi16sp x2, 16
+    mv x9, x2
+  )");
+  EXPECT_EQ(getGeneralRegister<uint64_t>(8), process_->getStackPointer());
+  EXPECT_EQ(getGeneralRegister<uint64_t>(9), process_->getStackPointer() + 16);
+}
+
+TEST_P(InstCompressed, slli) {
```

Comment:
Why do the test comments stop here?

Reply:
Could very well be GitHub not showing the changes correctly but thought I'd double check

Reply:
I didn't do these as thought they would be self explanatory. Have been added for completeness now